### PR TITLE
Paw 95

### PR DESCRIPTION
**Pull request by erwinXZ**
## CHECKLIST
**Sigue el checklist para ayudar a que tu PR se revice mas facil. Marca los puntos que ya completaste.**

- [x] Revisaste tu propio PR?

- [x] Agregaste una BREVE descripcion de este PR abajo.

- [x] Agregaste una descripcion de la SOLUCION de este PR abajo.

- [ ] Los cambios se aprobaron por diseño?

- [ ] Este PR paso QA?

## SUMMARY

Users Should be able to search and sort values by columns in the associations list

## SOLUTION

A directive was created to handle the sort of recrods and search inputs were added with functionallity to filter data results

### CODE REVIEW

File: src/app/private/admin/association-management/association-management.component.scss [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
   color: #fffdfd;
 
-  text-shadow: 0px 3px 4px rgba(0, 0, 0, 0.2);
+  text-shadow: 0px 3px 4px rgba(0, 0, 0, 0.2);
```
Comment (Amancaya) [inline src/app/private/admin/association-management/association-management.component.scss:19]: NIT: Can we revert this change? seems to add a no need it space

File: src/app/private/admin/association-management/association-management.component.ts [hunk 2]
```diff
@@ -1,33 +1,113 @@
...
+  sortByColumn = {};
+  sort = faSort;
+  sortAsc = faSortAsc;
+  sortDesc = faSortDesc;
+
+  constructor(private fb: FormBuilder, private associationService: AssociationService, private activatedRoute: ActivatedRoute, private router: Router, private modalService: NgbModal, private renderer: Renderer2, private targetElem: ElementRef) { }
 
   ngOnInit(): void {
+    const sort = new Sort();
+    this.sortByColumn = { name: { sortIcon: faSortAsc }, contactName: { sortIcon: faSort }, email: { sortIcon: faSort }, cellphone: { sortIcon: faSort } };
```
Comment (Amancaya) [inline src/app/private/admin/association-management/association-management.component.ts:42]: NIT: Can we be more descriptive with these variables `faSortAsc`, `faSort`

File: src/app/private/admin/association-management/association-management.component.ts [hunk 3]
```diff
@@ -1,33 +1,113 @@
...
+
+  onChanges(): void {
+    this.searchValues.valueChanges.subscribe(() => {
+      this.search();
+    });
   }
 
+  search(): void {
+    this.showLoader = true;
+    if (this.searchValues.get(['searchRefugeName']).value || this.searchValues.get(['searchContactName']).value || this.searchValues.get(['searchContactEmail']).value || this.searchValues.get(['searchContactCellphone']).value) {
```
Comment (Amancaya) [inline src/app/private/admin/association-management/association-management.component.ts:62]: It seems that these `'searchRefugeName'` is called on different places, since it's a string I wonder if we can handle it with as a constant, same for the other ones.

File: src/app/shared/util/sort.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+  constructor() {
+
+  }
+
+  public startSort(property, order, type = "") {
+    if (order === "desc") {
+      this.sortOrder = -1;
+    }
+    return (a, b) => {
```
Comment (Amancaya) [inline src/app/shared/util/sort.ts:16]: NIT: Can we be more descriptive with these variables a, b.

File: src/app/shared/util/sort.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        return this.sortData(new Date(a[property]), new Date(b[property]));
+      } else if (type === "value") {
+        return order == a[property] ? -1 : 0;
+      } else {
+        return this.collator.compare(a[property], b[property]) * this.sortOrder;
+      }
+    }
+  }
+
+  private sortData(a, b) {
```
Comment (Amancaya) [inline src/app/shared/util/sort.ts:27]: same here as above.

File: src/app/private/admin/association-management/association-management.component.ts (hunk 2 above)
Comment (erwinXZ) [reply src/app/private/admin/association-management/association-management.component.ts:42]: These are the name variables that Font Awesome uses we should keept ip since is a name convention coming from the library

File: src/app/private/admin/association-management/association-management.component.ts (hunk 3 above)
Comment (erwinXZ) [reply src/app/private/admin/association-management/association-management.component.ts:62]: Since the scope of that value is just on that method I recommed to keep it similar like the other values of searchContactName

**Review by Amancaya — APPROVED** [review]
